src/components/common/YourCompany/YourCompany.vue

```diff
@@ -466,10 +468,8 @@ export default class YourCompany extends Mixins(
       return this.getResource.changeData.nameChangeOptions
         .filter(option => option !== CorrectionTypes.CORRECT_NAME_TO_NUMBER)
     }
-    if (this.isEntityTypeFirm) {
-      return this.getResource.changeData.nameChangeOptions
-        .filter(option => option === CorrectionTypes.CORRECT_NEW_NR)
-    }
+    if (this.isEntityTypeSP) return SoleProprietorshipResource.changeData.nameChangeOptions
+    if (this.isEntityTypeGP) return GeneralPartnershipResource.changeData.nameChangeOptions
```

Comment:
Are the resource objects already available in the store? Search for setResource and getResource.

Reply:
Ditto in mixin below.